spring-integration-file/src/main/java/org/springframework/integration/file/remote/gateway/AbstractRemoteFileOutboundGateway.java

```diff
@@ -573,7 +608,7 @@ private void populateBeanFactoryIntoComponentsIfAny() {
 	private void setupLocalDirectory() {
 		File localDirectory =
 				ExpressionUtils.expressionToFile(this.localDirectoryExpression,
-						ExpressionUtils.createStandardEvaluationContext(getBeanFactory()), null,
+						this.standardEvaluationContext, null,
```

Comment:
If we do this, then `doInit()` has to be fixed respectively.
Pay attention to its logic: this `setupLocalDirectory()` method is called before you initialize `this.standardEvaluationContext`